number_theory/prime.h

```diff
@@ -0,0 +1,27 @@
+#ifndef NUMBER_THEORY_PRIME_H_
+#define NUMBER_THEORY_PRIME_H_
+
+#include <limits>
+
+// This file contains functions related to prime numbers.
+
+namespace number_theory {
+
+// Returns true if |number| is prime.
+template <class T>
+bool is_prime(T number) {
```

Comment:
This function can be marked as `constexpr`, so that the compiler can do more optimizations.

Reply:
I think saying `Otherwise, it returns false.` is just repeating the obvious fact. There are some functions that explicitly says that in chromium code base such as [this](https://source.chromium.org/chromium/chromium/src/+/main:base/files/file_path.h;l=477?q=file_path.h&ss=chromium%2Fchromium%2Fsrc) one.

However, more often than not, they just explain one of the cases which is usually the condition of `true`:

- https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/ash/arc/file_system_watcher/arc_file_system_watcher_util.h;l=43
- https://source.chromium.org/chromium/chromium/src/+/main:base/files/file_path.h;l=213?q=file_path.h&ss=chromium%2Fchromium%2Fsrc
- https://source.chromium.org/chromium/chromium/src/+/main:base/files/file_path.h;l=233?q=file_path.h&ss=chromium%2Fchromium%2Fsrc
- https://source.chromium.org/chromium/chromium/src/+/main:base/files/file_path.h;l=332?q=file_path.h&ss=chromium%2Fchromium%2Fsrc
- https://source.chromium.org/chromium/chromium/src/+/main:base/files/file_path.h;l=356?q=file_path.h&ss=chromium%2Fchromium%2Fsrc
- https://source.chromium.org/chromium/chromium/src/+/main:base/files/file_path.h;l=360?q=file_path.h&ss=chromium%2Fchromium%2Fsrc

There are many more examples like above in `file_path.h` file. I also noticed that sometimes the return value is not explicitly explained as in [this](https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/ash/arc/file_system_watcher/arc_file_system_watcher_util.h;l=28) case. I think the reason for this one is that function is not checking something, it tries to do an operation which is explained in its comment. Thus, return type being `bool` means that it returns `true` on success which might be better to explicitly mention though. It was my own commit, maybe it was overlooked somehow during the code review process. However, I think mentioning one of the conditions is usually more than enough.

Reply:
Done.